src/functions/bitcoin/psbt-functions.ts

```diff
@@ -56,25 +57,174 @@ export async function createFundingTransaction(
     network: bitcoinNetwork,
   });
 
-  const fundingTX = selected?.tx;
+  if (!selected) {
+    throw new Error(
+      'Failed to select Inputs for the Funding Transaction. Ensure sufficient funds are available.'
+    );
+  }
+
+  const fundingTX = selected.tx;
 
   if (!fundingTX) throw new Error('Could not create Funding Transaction');
 
   fundingTX.updateInput(0, {
     sequence: 0xfffffff0,
   });
 
-  const fundingPSBT = fundingTX.toPSBT();
+  return fundingTX;
+}
+
+/**
+ * Creates a Deposit Transaction.
+ * Uses the existing Vault's Funding Transaction ID and additional UTXOs to create the Deposit Transaction.
+ * The specified amount of Bitcoin is sent to the Vault's Multisig Address.
+ * The remaining amount is sent back to the user's address.
+ *
+ * @param bitcoinBlockchainURL - The Bitcoin Blockchain URL.
+ * @param bitcoinNetwork - The Bitcoin Network to use.
+ * @param depositAmount - The Amount of Bitcoin to deposit.
+ * @param vaultTransactionID - The ID of the Vault Funding Transaction.
+ * @param multisigPayment - The Taproot Multisig Payment Transaction.
+ * @param depositPayment - The User's Native Segwit or Taproot Payment Transaction which will be used to fund the Deposit Transaction.
+ * @param feeRate - The Fee Rate to use for the Transaction.
+ * @param feePublicKey - The Fee Recipient's Public Key.
+ * @param feeBasisPoints - The Fee Basis Points.
+ * @returns The Deposit Transaction.
+ */
+export async function createDepositTransaction(
+  bitcoinBlockchainURL: string,
+  bitcoinNetwork: Network,
+  depositAmount: bigint,
+  vaultTransactionID: string,
+  multisigPayment: P2TROut,
+  depositPayment: P2TROut | P2Ret,
+  feeRate: bigint,
+  feePublicKey: string,
+  feeBasisPoints: bigint
+): Promise<Transaction> {
+  const multisigPaymentAddress = multisigPayment.address;
+
+  if (!multisigPaymentAddress) {
+    throw new Error('Multisig Payment is missing Address');
+  }
+
+  const depositPaymentAddress = depositPayment.address;
+
+  if (!depositPaymentAddress) {
+    throw new Error('Deposit Payment is missing Address');
+  }
+
```

Comment:
You already have error checking that these aren't null before you call this function, IIRC. Probably better to have it only here, rather than in both places.

Reply:
Yes, removed it in the outer functions!